src/query/anonymization.c

```diff
@@ -428,6 +431,22 @@ static seed_t prepare_bucket_seeds(Query *query, ParamListInfo bound_params)
   return sql_seed;
 }
 
+static bool is_datetime_type(Oid type)
+{
+  switch (type)
+  {
+  case DATEOID:
+  case TIMEOID:
+  case TIMETZOID:
+  case TIMESTAMPOID:
+  case TIMESTAMPTZOID:
+    return true;
+  default:
+    /* Slower to evaluate, safety net in case new types get added. */
+    return TypeCategory(type) == TYPCATEGORY_DATETIME;
```

Comment:
Is this optimization really worth it?
We are calling this function from other locations as well and I haven't noticed an issue with it until now...

Reply:
I thought for a moment it is, when I realized it will be called a lot, and that it is quite heavy. OTOH, we don't yet have a convenient way to profile the extension set up, so once we get to performance tweaking, this might be harder to find. But on second thought, you're absolutely right, lets keep it simple..